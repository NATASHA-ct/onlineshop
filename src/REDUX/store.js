/* eslint-disable */
import { configureStore } from "@reduxjs/toolkit";
import productReducer from "./products/productsSlice";
import productDetailsReducer from "./products/productsdetailsSlice";

const store = configureStore({
  reducer: {
    products: productReducer,
    productDetail: productDetailsReducer,
  },
});

export default store;
