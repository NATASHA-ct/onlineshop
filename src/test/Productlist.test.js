import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../REDUX/store';
import ProductList from '../COMPONENTS/PRODUCTSLIST/Productlist';

describe('to test the ProductList components', () => {
  test('matches the ProductList DOM snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <ProductList />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
