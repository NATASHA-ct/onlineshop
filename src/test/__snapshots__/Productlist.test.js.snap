// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to test the ProductList components matches the ProductList DOM snapshot 1`] = `
<div>
  <div
    className="input"
  >
    <p
      className="search-title"
    >
      Search below for your favorite product..
    </p>
    <input
      name="search"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <h2
    className="list-heading"
  >
    AVAILABLE PRODUCTS
  </h2>
  <div
    className="list-container"
  />
</div>
`;

exports[`to test the Productcard components matches the Productcard DOM snapshot 1`] = `
<a
  href="/product/undefined"
  onClick={[Function]}
>
  <div
    className="card"
  >
    <div
      className="card-container"
    >
      <img
        className="card-img"
        height="200px"
        width="200px"
      />
      <div
        className="card-details"
      >
        <h1
          className="card-title"
        />
        <p
          className="card-price"
        >
          $ 
        </p>
      </div>
    </div>
  </div>
</a>
`;
